components/script/dom/webidls/TextTrack.webidl

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+// https://html.spec.whatwg.org/multipage/#texttrack
+
+enum TextTrackMode { "disabled",  "hidden",  "showing" };
+enum TextTrackKind { "subtitles",  "captions",  "descriptions",  "chapters",  "metadata" };
+
+[Exposed=Window]
+interface TextTrack : EventTarget {
+  readonly attribute TextTrackKind kind;
+  readonly attribute DOMString label;
+  readonly attribute DOMString language;
+
+  readonly attribute DOMString id;
+  // readonly attribute DOMString inBandMetadataTrackDispatchType;
+
+  attribute TextTrackMode mode;
+
+  readonly attribute TextTrackCueList? cues;
+  // readonly attribute TextTrackCueList? activeCues;
+
+  [Throws]
```

Comment:
Note the [IDL](https://html.spec.whatwg.org/multipage/media.html#texttrack) does not include `Throws`, but [addCue](https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-addcue) can throw a "InvalidStateError" in step 2 and [removeCue](https://html.spec.whatwg.org/multipage/media.html#dom-texttrack-removecue) can throw a  "NotFoundError" in step 1.

Reply:
Yeah, [Throws] is a Gecko/Servo-specific extension.

Reply:
:+1: Thanks